src/ServiceDiscovery/Services.cpp

```diff
@@ -444,10 +439,87 @@ bool Services::SendTemporaryROOTplot(const std::string& plot_name, const std::st
   
 }
 
-bool Services::SendPlot(){
-  
-  // placeholder for evgenii
-  
+static std::vector<float> decodePostgresArray(const std::string& string) {
+  if (string.length() < 2) return std::vector<float>(); // "{}"
+  size_t length = 1;
+  for (size_t i = 0; i < string.size(); ++i) if (string[i] == ',') ++length;
+
+  std::vector<float> result(length);
+
+  std::stringstream ss;
+  ss << string;
+  size_t i = 0;
+  char c;
+  ss >> c; // '{'
+  while (ss && c != '}' && i < length) ss >> result[i++] >> c;
+  return result;
+};
+
+static std::string encodePostgresArray(const std::vector<float>& array) {
```

Comment:
is ths postgress specific?? or is it jsut a text encoding so can be called encode array?

same with decode. Would be good to be agnosti to backend, so references to postgress should be removed